### PR TITLE
Misc clean up for PinotHelixResourceManager

### PR DESCRIPTION
**Pull request by Jackie-Jiang**

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -1321,19 +1304,15 @@ public void updateIndexingConfigFor(String tableName, TableType type, IndexingCo
...
-          @Override
-          public IdealState apply(@Nullable IdealState idealState) {
-            Preconditions.checkNotNull(idealState);
-            for (String broker : brokersForTenant) {
-              idealState.setPartitionState(tableName, broker, BrokerOnlineOfflineStateModel.ONLINE);
-            }
-            return idealState;
-          }
-        }, RetryPolicies.exponentialBackoffRetryPolicy(5, 500L, 2.0f));
+  private void handleBrokerResource(String tableNameWithType, List<String> brokersForTenant) {
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: suggest rename to enableTableInBrokers(). Or, updateBrokerResource() but that is more vague.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -394,44 +387,40 @@ public synchronized PinotResourceManagerResponse addInstance(@Nonnull Instance i
...
-  public List<String> getAllRealtimeTables() {
-    List<String> resourceNames = getAllResources();
-    Iterator<String> iterator = resourceNames.iterator();
-    while (iterator.hasNext()) {
-      if (!TableNameBuilder.REALTIME.tableHasTypeSuffix(iterator.next())) {
-        iterator.remove();
+  public List<String> getAllOfflineTables() {
+    List<String> offlineTableNames = new ArrayList<>();
+    for (String resourceName : getAllResources()) {
+      if (TableNameBuilder.OFFLINE.tableHasTypeSuffix(resourceName)) {
```
Comment (jackjlli) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: I prefer keeping this method `tableHasTypeSuffix` within TableNameBuilder and marked as private access method. 
And this line be replaced by `TableNameBuilder.isOfflineTableResource(resourceName)`. 😉

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -394,44 +387,40 @@ public synchronized PinotResourceManagerResponse addInstance(@Nonnull Instance i
...
-  public List<String> getAllOfflineTables() {
-    List<String> resourceNames = getAllResources();
-    Iterator<String> iterator = resourceNames.iterator();
-    while (iterator.hasNext()) {
-      if (!TableNameBuilder.OFFLINE.tableHasTypeSuffix(iterator.next())) {
-        iterator.remove();
+  public List<String> getAllRealtimeTables() {
+    List<String> realtimeTableNames = new ArrayList<>();
+    for (String resourceName : getAllResources()) {
+      if (TableNameBuilder.REALTIME.tableHasTypeSuffix(resourceName)) {
```
Comment (jackjlli) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Same here. `TableNameBuilder.isRealtimeTableResource(resourceName)` might be better.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Inline it into the addTable() method to be more clear

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done

**Pull request merged**